design/ErrorHandling.md

```diff
@@ -50,6 +51,13 @@ The device and all objects created from the device have become invalid.
 All further operations on the device and its objects are errors.
 The `"validationerror"` event will no longer fire. (This makes all further operations no-ops.)
 
+An app should never give up on getting WebGPU access due to
+`requestDevice` returning null or `GPUDevice.lost` resolving.
+It should only give up based on a `requestAdapter` rejection.
+(It should also give up on a `requestDevice` rejection, as that indicates an app programming error.)
+If `requestDevice` rejects, either the app has requested capabilities the adapter doesn't have,
+or the adapter has been lost and the app should `requestAdapter` again.
```

Comment:
This sentence seems to contradict text above that says that if the adapter is lost then the promise returns null.

Reply:
oops, rebase error